### PR TITLE
bug: triage role scans all repo issues due to empty label in BuildStateFromGitHub

### LINKED ISSUE #133: bug: triage role scans all repo issues due to empty label in BuildStateFromGitHub
## Bug

The triage role accidentally recovers unrelated issues on startup because `BuildStateFromGitHub` runs a labeled-issue scan with an empty label, which the GitHub API treats as "no filter" — returning all open issues in the repo.

## Root cause

In `BuildRoleEntries` (`fileconfig.go:370-388`), the triage role never sets `cfg.Label`:

```go
for _, triage := range p.Triage {
    cfg := baseCfg
    cfg.TriageJobs = triage.Jobs
    // ... no cfg.Label assignment
}
```

Then in `BuildStateFromGitHub` (`state.go:50-51`), the labeled-issue scan runs because `WatchPRs` is empty:

```go
if len(cfg.WatchPRs) == 0 {
    issues, err := gh.ListLabeledIssues(ctx, cfg.Owner, cfg.Repo, cfg.Label) // label = ""
}
```

`ListLabeledIssues` passes `Labels: []string{""}` to the GitHub API, which returns **all open issues**. Any issue with the `ai-failed` label is then recovered into state — even if it was created by a different oompa instance or has nothing to do with this triage role.

## Observed behavior

```
level=INFO msg="recovered failed issue from GitHub" role=triage issue=2692
level=INFO msg="recovered failed issue from GitHub" role=triage issue=2680
level=INFO msg="recovered failed issue from GitHub" role=triage issue=2614
```

None of these issues have the triage role's configured label — they only have `ai-failed` from a previous (unrelated) oompa run.

## Expected behavior

The triage role should **not** scan for labeled issues at all. It only processes CI job URLs — it never creates or manages issues through the label-based workflow.

## Proposed fix

Skip the labeled-issue scan in `BuildStateFromGitHub` when `cfg.Label` is empty:

```go
// state.go:50
if len(cfg.WatchPRs) == 0 && cfg.Label != "" {
    issues, err := gh.ListLabeledIssues(ctx, cfg.Owner, cfg.Repo, cfg.Label)
    // ...
}
```

This is safe because:
- The `issues` role always has a label set (defaults to `good-for-ai`)
- The `prs` role skips this block via `len(cfg.WatchPRs) > 0`
- The `triage` role has no label and should skip entirely

### Files to modify

| File | Change |
|---|---|
| `pkg/agent/state.go:50` | Add `cfg.Label != ""` guard to the labeled-issue scan |
| `pkg/agent/state_test.go` | Add test: `TestBuildStateFromGitHub_EmptyLabelSkipsScan` |

### PR DESCRIPTION
**Pull request by qinqon**
Fixes #133

## Summary

Fix triage role accidentally scanning all repo issues due to empty label in `BuildStateFromGitHub`. When `cfg.Label` is empty, `ListLabeledIssues` passes `Labels: []string{""}` to the GitHub API, which returns all open issues -- causing unrelated `ai-failed` issues to be recovered into state.

## Changes

- Add `cfg.Label != ""` guard to the labeled-issue scan condition in `BuildStateFromGitHub` (`pkg/agent/state.go:50`)
- Add `TestBuildStateFromGitHub_EmptyLabelSkipsScan` test verifying that an empty label config produces empty state

## Testing

- [x] `go test ./...` passes
- [x] `go vet ./...` passes
- [ ] Manual testing (if applicable)

## Related Issues

Fixes #133

### CODE REVIEW

**Comment by qinqon** [comment]
Rebased commit 1b36fbc on main and pushed.

File: pkg/agent/state.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ func BuildStateFromGitHub(ctx context.Context, gh GitHubClient, cfg Config, clon
 
 	// Skip labeled issue scanning when --watch-prs is configured
 	// (watch mode bypasses issue discovery)
-	if len(cfg.WatchPRs) == 0 {
+	if len(cfg.WatchPRs) == 0 && cfg.Label != "" {
 		issues, err := gh.ListLabeledIssues(ctx, cfg.Owner, cfg.Repo, cfg.Label)
```
[the remark this answers is not in the document]
Comment (qinqon) [reply pkg/agent/state.go:52]: > `BuildStateFromGitHub` now skips labeled-issue discovery when `cfg.Label == ""`, but the agent's runtime poll loop still calls `ProcessNewIssues()`, which calls `ListLabeledIssues()` with `a.cfg.Label` unconditionally.

Addressed: Added an early return in `ProcessNewIssues` when `a.cfg.Label` is empty, preventing the runtime poll loop from calling `ListLabeledIssues` with an empty label. Also updated the comment in `BuildStateFromGitHub` to document both skip conditions, and added `TestProcessNewIssues_EmptyLabelSkips` verifying that `ListLabeledIssues` is not called when the label is empty.

[the remark this answers is not in the document]
Comment (qinqon) [reply pkg/agent/state.go:52]: Addressed in the latest push.

**Comment by qinqon** [comment]
Rebased commit e7fc3ff on main and pushed.
